### DISCUSSION TITLE
Thank You for the MSIX Upload Action!

### DISCUSSION THREAD

**Discussion by eythaann** [JasonWei512/Upload-Microsoft-Store-MSIX-Package-to-GitHub-Release#8] [General]
Hi Jason,

I recently started using your action to upload MSIX packages to GitHub Releases in my repository, [Seelen-UI](https://github.com/eythaann/Seelen-UI), and I wanted to take a moment to thank you for creating and sharing this tool. It's been incredibly helpful and has streamlined my workflow significantly. Your work is greatly appreciated!

Additionally, combining this action with the [Winget Releaser](https://github.com/marketplace/actions/winget-releaser) action to upload the MSIX to Winget makes for a very useful and powerful combination.

As a suggestion, it would be fantastic if instead of checking every 6 hours (as the example action), the action could respond to notifications from Microsoft when the MSIX bundle has been accepted and uploaded to the store. Typically, Microsoft sends an email for such updates, and there might be a hook in the Partner Center API for listening to the "on submit completed" event. Perhaps you could look into the code of [StoreBroker](https://github.com/microsoft/StoreBroker/blob/master/Documentation/USAGE.md#monitoring-a-submission) or [msstore-cli](https://github.com/microsoft/msstore-cli) for some ideas.

Best regards,  
Eythan

**Comment by JasonWei512** [comment]
Hi eythaann,

I'm glad you like this action!

About the notification trigger:

- [StoreBroker](https://github.com/microsoft/StoreBroker/blob/master/Documentation/USAGE.md#monitoring-a-submission) monitors submissions by polling the status every 60 seconds.
  
  I haven't used it, but I guess that's not very different from the current "checking every 6 hours" approach?

  If 6 hours is too long for you, you can let the action run every 5 minutes, which is the the shortest interval allowed by GitHub, according to the [documentation](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule).

- Microsoft Partner Center does have a [Subscription Updated Event](https://learn.microsoft.com/en-us/partner-center/developer/partner-center-webhooks) webhook. 
  
  I haven't tested it, so I am not sure if it's the right one to use. But according to the documentation, there is a delay of up to 48 hours between the time a subscription changes and when this event is triggered.

I am currently working on some other projects, so I may not have time to work on this GitHub action. If you would like to create a pull request, I would be very grateful.

And again, thank you for using this action!
